### ISSUE TITLE
[Feature Request] Pdf-docx export option

### ISSUE THREAD

**Issue by libantas** [zadam/trilium#1387]
It would be a must have a good pdf - docx export option. At the end, the documents of most of the people are in docx and pdf.

**Comment by zDEFz** [comment]
hard against. You open Pandoras Box. Then you'd also need to support OpenDocument Standards like...
ODT...

**Comment by cebas** [comment]
this could be a good incentive to implement a plugin system that allows, among other stuff, to import/export from/to whatever format the plugin implementer wants
this will give more use cases to Trilium and more users and community support

**Comment by zDEFz** [comment]
I agree with @cebas, but don't forget that this is really much todo.
What would that mean? 

Is this about documents only?
Is this also for multi-media? Like movies, music etc? If so, implementation of FFMPEG might be problematic for some companies, cause it might give you licensing problems.

**Comment by zadam** [comment]
This is sort of feature creep - these are not simple features and are only tangentially related to the focus of Trilium - note taking.

For docx there's an easy workaround - you can just open Word and paste the note's content into it and there you go - docx.

**Comment by cebas** [comment]
> I agree with @cebas, but don't forget that this is really much todo.
> What would that mean?
> 
> Is this about documents only?
> Is this also for **multi-media? Like movies, music etc?** **If so, implementation of FFMPEG might is problematic for some companies, cause it might give you licensing problems.**

Not only documents. My thoughts were about a generic plugin system that allows any kind of extension that the plugin system allows. This could start with import/export and grow to every other way @zadam wants to open a door to.

**Comment by zDEFz** [comment]
I think you must explain more. This sounds to me, that you could do that by passing arguments to another executable file on windows / mac / linux.

**Comment by cebas** [comment]
I mean this: https://en.wikipedia.org/wiki/Plug-in_(computing)

**Comment by zadam** [comment]
@cebas in theory Trilium already has "plugins" - they are called "scripts". Trilium scripting API provides you access to the database and you can use that to export in whatever file format you wish.

There's a catch though - I doubt people will develop docx export from scratch and will want to use existing node.js libraries. And this is a problem since the set of libraries Trilium comes with is set at build time and cannot be extended. Trilium scripts are CommonJS-like modules and in theory some libraries could be shoe horned into scripts, I'm not sure how this would work in practice ...

**Comment by cebas** [comment]
@zadam thanks for the explanation. I didn't notice these scripts are plugin-like. I have no experience with node, so I can't help on this, but I'm sure somebody else will.
You may need to find a way to develop scripts with extra node libraries. This may be needed to create complex plugins as the docx.

**Comment by wwwald** [comment]
There is always [Pandoc](https://pandoc.org/) for flexible file format conversion, based on Trilium's current HTML or Markdown export. See the [demos page](https://pandoc.org/demos) for example conversions, it includes docx as export option.

**Comment by zDEFz** [comment]
I just checked the demo page and it doesn't convert files for me...

**Comment by wwwald** [comment]
Any error messages? Are you trying to execute on an HTML or Markdown export of Trilium?

Willing to help out, Pandoc has usually worked well for me over the past years.

**Comment by zDEFz** [comment]
Nah, not at all. I thought I can import my files into that thing and get a PDF.
There was no function to import data and export data, just to parse for instance Markdown to HTML...

I mean, you are meant to renderNote the HTML file from the demo, and that's all you can use, no?

If it's different, write step-by-step please.

**Comment by eliandoran** [comment]
Implemented export as PDF in [TriliumNext](https://github.com/TriliumNext/Notes/pull/1091).
